packages/core/src/storage/defaultProxyStorage.ts

```diff
@@ -0,0 +1,114 @@
+import type { MessageType, StorageThreadType } from '../memory/types';
+import { MastraStorage } from './base';
+import type { TABLE_NAMES } from './constants';
+import type { DefaultStorage, type LibSQLConfig } from './libsql';
+import type { EvalRow, StorageColumn, StorageGetMessagesArg } from './types';
+
+/**
+ * A proxy for the DefaultStorage (LibSQLStore) to allow for dynamically loading the storage in a constructor
+ */
+export class DefaultProxyStorage extends MastraStorage {
+  private storage: DefaultStorage | null = null;
+  private storageConfig: LibSQLConfig;
+
+  constructor({ config }: { config: LibSQLConfig }) {
+    super({ name: 'DefaultStorage' });
+    this.storageConfig = config;
+  }
+
+  private async setupStorage() {
+    const { DefaultStorage } = await import('./libsql');
+
+    this.storage = new DefaultStorage({ config: this.storageConfig });
```

Comment:
we need an init guard right? looks like this will get re-initialized in every method call

Reply:
Also curious why we don't make these changes inside the libsql class directly - users always need to call an async method before interacting with it so we should be able to do the same inside it right?

Reply:
need to guard against init race conditions too https://github.com/mastra-ai/mastra/blob/main/packages/core/src/storage/base.ts#L152

Reply:
@TylerBarnes what race condition are you talking about? we do have the promise set?

Reply:
ah i get it, fuck good catch